### PR TITLE
feat(1530): changed to remote couchdb with caching

### PR DESCRIPTION
**Pull request by akhilmhdh**
# Closes
1. 1530

# Explain the feature/fix

1. Changed to remote couchdb
2. Implemented caching
3. Fixed ui of comment submit
4. Fixed change feed issue

## Does this PR introduce a breaking change

Yes

## Screenshots

<details>
<summary>View Screenshots</summary>

Except comment box no ui changes

| Description | Screenshot |
| - | - |
| Comment Box | Screenshot 2021-09-27 at 3 10 31 PM
> |

</details
### Ready-for-merge Checklist

- [x] Expected files: all files in this pull request are related to one feature request or issue (no stragglers)?
- [ ] Does the change have appropriate unit tests?
- [ ] Did tests pass?
- [x] Did you update or add any necessary documentation (README.md, WHY.md, etc.)?
- [x] Was this feature demo'd and the design review approved?

### CODE REVIEW

**Review by riginoommen — APPROVED** [review]

File: src/pages/HomePage/hooks/useGetTagCountList.ts [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+    mutate: mutateTags,
+  } = useSWR('/tag-counts', async (): Promise<PouchDB.Query.Response<TagDoc>['rows']> => {
+    return new Promise((resolve) => {
+      window.OpAuthHelper.onLogin(async () => {
+        return resolve(handleFetchTags());
+      });
+    });
+  });
+
+  const isTagsLoading = !error && !tagCount;
```
Comment (ghanlohar) [inline src/pages/HomePage/hooks/useGetTagCountList.ts:39]: Use Boolean() for this for readability.

**Review by ghanlohar — CHANGES_REQUESTED** [review]

File: src/pages/HomePage/hooks/useGetTagCountList.ts [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+};
+
+const handleFetchTags = async (): Promise<PouchDB.Query.Response<TagDoc>['rows']> => {
+  try {
+    const { rows } = await tagDoc.getTagCounts();
+    rows.sort((a, b) => b.value - a.value).splice(10);
+    return rows;
+  } catch (error) {
+    window.OpNotification.danger({
+      subject: `Failed to fetch tags`,
```
[the remark this answers is not in the document]
Comment (akhilmhdh) [reply src/pages/HomePage/hooks/useGetTagCountList.ts:18]: Good catch!!

File: src/containers/CommentsContainer/hooks/useGetComments.tsx [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+    },
+    { revalidateOnFocus: false }
+  );
+  const commentsRef = useRef(comments);
+
+  useEffect(() => {
+    commentsRef.current = comments;
+  }, [comments]);
+
+  const isCommentsLoading = !comments && !error;
```
[the remark this answers is not in the document]
Comment (akhilmhdh) [reply src/containers/CommentsContainer/hooks/useGetComments.tsx:66]: @ghanlohar  Basically that would fire eslint error of redundancy.....i can change it to `!(comments && error)`. But i felt this makes more sense on reading like no comments and no error means loading

[the remark this answers is not in the document]
Comment (akhilmhdh) [reply src/containers/CommentsContainer/hooks/useGetComments.tsx:66]: Or did u mean vice versa...Boolean(!data) that would also fire redundant eslint error

[the remark this answers is not in the document]
Comment (ghanlohar) [reply src/containers/CommentsContainer/hooks/useGetComments.tsx:66]: Boolean(!data) would be lint-free but I think it's fine using it as it is.

**Review by ghanlohar — DISMISSED** [review]

File: src/containers/CommentsContainer/components/CommentBox/CommentBox.tsx [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+
+  return (
+    <CommentField commenterName={author} createdAt={createdAt}>
+      <Text aria-label="comment-content">{content}</Text>
+      <CommentField.CommentButton onClick={onLikeButtonClick} isDisabled={isLiking}>
+        {formatCommentLikeButton(hasLiked, votes, isLiking)}
+      </CommentField.CommentButton>
+      {/* <CommentField.CommentButton onClick={() => handleReplyButtonClick(authorId)}>
+              Reply
+            </CommentField.CommentButton> */}
```
Comment (deshmukhmayur) [inline src/containers/CommentsContainer/components/CommentBox/CommentBox.tsx:47]: Remove unused / commented statements.

**Review by deshmukhmayur — CHANGES_REQUESTED** [review]

File: src/containers/CommentsContainer/CommentsContainer.tsx [hunk 5]
```diff
@@ -159,26 +180,22 @@ export const CommentsContainer = ({ ideaDetails }: Props): JSX.Element => {
...
-      commentArea.current?.focus();
-    }
-  }, []);
+  // const handleReplyButtonClick = useCallback((replyTo: string) => {
+  //   if (commentArea.current) {
+  //     commentArea.current.value = `@${replyTo} `;
+  //     window.scrollTo({ top: 0, behavior: 'smooth' });
+  //     commentArea.current?.focus();
+  //   }
+  // }, []);
```
[the remark this answers is not in the document]
Comment (akhilmhdh) [reply src/containers/CommentsContainer/CommentsContainer.tsx]: Done

**Review by deshmukhmayur — APPROVED** [review]

**Review by riginoommen — APPROVED** [review]

**Pull request merged**